### PR TITLE
Feat/Custom Policy Assignment -- Feature Complete

### PR DESCRIPTION
**Pull request by GilTeraSky**
1. **What this PR does / why we need it**:
    This PR includes the complete implementation of a Custom Policy Assignment.
2. **Which issue(s) this PR fixes**

3. **Additional information**
    a. It was needed to modify the InputData struct to support custom policy with custom template.
    b. Parameters field is a string because it is a dynamic field, therefore it will be populate with JSON encoded string for the 
        parameters.
    c. On plan phase, parameters will be validated according to the OpenAPIV3 Schema that is defined for the custom template.

4. **Special notes for your reviewer**

### CODE REVIEW

File: internal/resources/policy/kind/custom/input_schema.go [hunk 1]
```diff
@@ -173,42 +203,27 @@ func ValidateInput(ctx context.Context, diff *schema.ResourceDiff, i interface{}
...
-		if recipeType, ok := recipeData.([]interface{}); ok && len(recipeType) != 0 {
-			recipesFound = append(recipesFound, reciperesource.TMCHTTPSIngressKey)
-		}
-	}
-
-	if recipeData, ok := inputData[reciperesource.TMCRequireLabelsKey]; ok {
-		if recipeType, ok := recipeData.([]interface{}); ok && len(recipeType) != 0 {
-			recipesFound = append(recipesFound, reciperesource.TMCRequireLabelsKey)
+	inputData, _ := inputType[0].(map[string]interface{})
+	recipes := []string{reciperesource.TMCBlockNodeportServiceKey, reciperesource.TMCBlockResourcesKey,
```
Comment (shobha2626) [inline internal/resources/policy/kind/custom/input_schema.go:209]: We can make use of the `RecipesAllowed` variable defined above?

File: internal/resources/policy/kind/custom/input_schema.go [hunk 2]
```diff
@@ -33,10 +45,10 @@ var (
 				reciperesource.TMCExternalIPSKey:              reciperesource.TMCExternalIps,
 				reciperesource.TMCHTTPSIngressKey:             reciperesource.TMCHTTPSIngress,
 				reciperesource.TMCRequireLabelsKey:            reciperesource.TMCRequireLabels,
+				reciperesource.TMCCustomKey:                   reciperesource.TMCCustomSchema,
```
Comment (shobha2626) [inline internal/resources/policy/kind/custom/input_schema.go:48]: Is acceptance test added for the new custom schema?

Please attach the screenshot of the same

**Review by shobha2626 — CHANGES_REQUESTED** [review]

**Comment by shobha2626** [comment]
Please squash the commits into meaningful commit message

File: internal/resources/policy/kind/custom/input_schema.go (hunk 2 above)
Comment (GilTeraSky) [reply internal/resources/policy/kind/custom/input_schema.go:48]: I didn't find acceptance tests for the custom policy resource in general, I did implement mock tests though.

**Review by shobha2626 — APPROVED** [review]

**Review by ankitsny — APPROVED** [review]

**Comment by ramya-bangera** [comment]
@shobha2626 - I could not find the acceptance tests , is it part of separate PR?

**Comment by shobha2626** [comment]
> @shobha2626 - I could not find the acceptance tests , is it part of separate PR?

@ramya-bangera acceptance test was not done as part of this PR. Will be opening a new PR with acceptance test changes

**Comment by shobha2626** [comment]
Added acceptance test as part of the PR - https://github.com/vmware/terraform-provider-tanzu-mission-control/pull/371

**Review by ramya-bangera — APPROVED** [review]

**Review by sbasu31 — APPROVED** [review]

**Pull request merged**